packages/manifold/contracts/physicalclaim/PhysicalClaimCore.sol

```diff
@@ -0,0 +1,337 @@
+     * Initialiazes a physical claim with base parameters
+     */
+    function _initialize(
+        uint256 instanceId,
+        PhysicalClaimParameters calldata physicalClaimParameters
+    ) internal {
+        _physicalClaimCreator[instanceId] = msg.sender;
+        PhysicalClaimLib.initialize(instanceId, _physicalClaims[instanceId], physicalClaimParameters);
+    }
+
+    /**
+     * Updates a physical claim with base parameters
+     */
+    function _update(
+        uint256 instanceId,
+        PhysicalClaimParameters calldata physicalClaimParameters
+    ) internal {
+        PhysicalClaimLib.update(instanceId, _getPhysicalClaim(instanceId), physicalClaimParameters);
+    }
+
+    /**
+     * Validates that this physical claim is managed by the user
+     */
+     function _validateAdmin(
+        uint256 instanceId
+     ) internal view {
+        require(_physicalClaimCreator[instanceId] == msg.sender, "Must be admin");
+     }
+
+    /**
+     * See {IPhysicalClaimCore-getPhysicalClaim}.
+     */
+    function getPhysicalClaim(uint256 instanceId) external override view returns(PhysicalClaim memory) {
+        return _getPhysicalClaim(instanceId);
+    }
+
+    /**
+     * Helper to get physical claim instance
+     */
+    function _getPhysicalClaim(uint256 instanceId) internal view returns(PhysicalClaim storage physicalClaimInstance) {
+        physicalClaimInstance = _physicalClaims[instanceId];
+        if (physicalClaimInstance.paymentReceiver == address(0)) {
+            revert InvalidInstance();
+        }
+    }
+
+    /**
+     * (Batch overload) see {IPhysicalClaimCore-burnRedeem}.
+     */
+    function burnRedeem(PhysicalClaimSubmission[] calldata submissions) external payable override nonReentrant {
+        if (submissions.length == 0) {
+            revert InvalidInput();
+        }
+        uint256 msgValueRemaining = msg.value;
+        for (uint256 i; i < submissions.length;) {
+            msgValueRemaining -= _burnRedeem(msgValueRemaining, submissions[i]);
+            unchecked { ++i; }
+        }
+
+        if (msgValueRemaining != 0) {
+            _forwardValue(payable(msg.sender), msgValueRemaining);
+        }
+    }
+
+    function _burnRedeem(uint256 msgValue, PhysicalClaimSubmission calldata submission) private returns (uint256) {
+        PhysicalClaim storage physicalClaimInstance = _getPhysicalClaim(submission.instanceId);
+
+        // Get the amount that can be burned
+        uint32 physicalClaimCount = _getAvailablePhysicalClaimCount(physicalClaimInstance.totalSupply, physicalClaimInstance.redeemedCount, submission.physicalClaimCount);
+
+        uint cost;
+        if (physicalClaimInstance.signer == address(0)) {
+            cost = 0;
+        } else {
+            // Check that the message value is what was signed...
+            _checkPriceSignature(submission.instanceId, submission.signature, submission.message, submission.nonce, physicalClaimInstance.signer, msgValue);
+        }
+
```

Comment:
Refactor for efficiency:
if (cost > 0) {
  if (physicalClaimCount > 1) {
    cost *= physicalClaimCount;
   }
  if (remainingValue < cost) {
    revert InvalidPaymentAmount();
  }
  remainingValue -= cost;
  _forwardValue(physicalClaimInstance.paymentReceiver, cost);
}

Reply:
Updated (bit different than you had)